source/plugin/permissions.rst

```diff
@@ -2,9 +2,229 @@
+
+Usage-Example
+~~~~~~~~~~~~~
+
+.. code-block:: text
+
+    Optional<Builder> optBuilder = permissionService.newDescriptionBuilder(myPlugin);
+    if (optBuilder.isPresent()) {
+        Builder builder = optBuilder.get();
+        builder.id("myPlugin.commands.teleport.execute")
+               .description(Text.of("Allows the user to execute the teleport command."))
+               .assign(PermissionDescription.ROLE_STAFF, true)
+               .register();
+    }
+
+Simple-Result
+~~~~~~~~~~~~~
+
+.. code-block:: text
+    
+    myPlugin.commands.teleport.execute
+    
+    Description: Allows the user to execute the teleport command.
+    Role: user
+    Owner: MyPlugin v1.2.3
+
+Template-result
+~~~~~~~~~~~~~~~
+
+.. code-block:: text
+    
+    myPlugin.commands.teleport.worlds.<World>
+    
+    Description: Allows the user to teleport to the world <World>.
+    Role: staff
+    Owner: MyPlugin v1.2.3
+
+.. tip::
+
+    You might skip writing descriptions for some parent permission groups such as ``myPlugin.commands.teleport.worlds``
+    or ``myPlugin.commands`` as their meaning can be derived from the permission structure and the defined children
+    alone.
+
+Subject
+=======
+
+A ``Subject`` is a holder of assigned permissions it can be obtained from the ``PermissionService`` via 
+``SubjectCollections``.
+``CommandSources`` such as ``Players`` are ``Subjects`` by default, but there are many other types of ``Subjects`` as
+well. Anything that has permissions is a Subject even if it just delegates the checks to a contained Subject.
+Permissions can be granted or denied to a Subject. If a permission is neither granted nor denied its setting will be
+inherited. See Inheritance.
+Subjects provide methods to check whether it has a certain permission or not.
+Plugins that use this method should only query for the specific permission they want to check. It is the
+PermissionService's task to respect the permission and subject inheritance.
+
+Example
+~~~~~~~
+
+The following example could be used to check whether the Player is allowed to execute the teleport command.
+
+.. code-block:: java
+    
+    public canTeleport(Player subject, World targetWorld) {
+        return subject.hasPermission("myPlugin.command.teleport.execute") 
+                && (subject.getWorld() == targetWorld
+                        || subject.hasPermission("myPlugin.command.teleport." + targetWorld.getName()));
+    }
+
+Inheritance
+~~~~~~~~~~~
+
+If a ``Subject`` got a permission assigned, it will use that value.
+Otherwise it will be inherited from any parent ``Subject``. It does not matter what kind of parent (ex group or player)
+``Subject`` that might be.
+
+If neither the subject itself nor any parent subjects grant or deny a permission then it will be inherited from the
+default ``Subject``. The default subject can be obtained from the ``PermissionService``.
```

Comment:
It can also be obtained from the SubjectCollection.

There are two layers of defaults. The SubjectCollection's defaults are to be considered before the PermissionService's global defaults.